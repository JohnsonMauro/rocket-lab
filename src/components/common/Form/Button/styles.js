import styled from '@emotion/styled';
import { css } from '@emotion/core';

const variantns = ({ variant, theme }) =>
  ({
    primary: css`
      background-color: ${theme.colors.blazeOrange};
      border: 1px solid ${theme.colors.blazeOrange};
      color: ${theme.colors.white};
    `,
    secondary: css`
      background-color: ${theme.colors.white};
      border: 1px solid ${theme.colors.white};
      color: ${theme.colors.blazeOrange};
    `,
  }[variant]);

const ButtonStyles = styled.button`
  ${variantns};
  border-radius: 1.8px;
  cursor: pointer;
  outline: none;
  padding: 7px 22px;
  user-select: none;
  vertical-align: middle;
`;

export { ButtonStyles };
