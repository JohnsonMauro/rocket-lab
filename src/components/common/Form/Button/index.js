import React from 'react';
import { ButtonStyles } from './styles';

const Button = ({ children, onClick, type, variant }) => (
  <ButtonStyles onClick={onClick} type={type} variant={variant}>
    {children}
  </ButtonStyles>
);

Button.defaultProps = {
  type: 'button',
  variant: 'primary',
};

export default Button;
