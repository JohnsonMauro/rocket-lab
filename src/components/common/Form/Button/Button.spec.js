import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import Button from './index';

describe('<Button />', () => {
  it('Renders the primary button component', () => {
    const { container } = render(<Button />, {
      wrapper: GlobalsContainer,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders the secondary button component', () => {
    const { container } = render(<Button variant="secondary" />, {
      wrapper: GlobalsContainer,
    });
    expect(container).toMatchSnapshot();
  });
});
