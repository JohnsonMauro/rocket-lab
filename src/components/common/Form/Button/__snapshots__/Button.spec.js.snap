// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Renders the primary button component 1`] = `
<div>
  <button
    class="css-1o9np2w-ButtonStyles-primary e10ohzgh0"
    type="button"
  />
</div>
`;

exports[`<Button /> Renders the secondary button component 1`] = `
<div>
  <button
    class="css-leptw7-ButtonStyles-secondary e10ohzgh0"
    type="button"
  />
</div>
`;
